templates/CRM/UF/Form/Fields.tpl

```diff
@@ -89,16 +89,17 @@
             {include file="CRM/Profile/Form/GreetingType.tpl"}
           {elseif ($profileFieldName eq 'group' && $form.group) || ($profileFieldName eq 'tag' && $form.tag)}
             {include file="CRM/Contact/Form/Edit/TagsAndGroups.tpl" type=$profileFieldName title=null context="profile"}
-          {elseif ( $profileFieldName|substr:-5:5 eq '_date' ) AND
-          ( $form.formName neq 'Confirm' )  AND
-          ( $form.formName neq 'ThankYou' ) }
-            {include file="CRM/common/jcalendar.tpl" elementName=$profileFieldName}
-          {elseif ( $profileFieldName|substr:-5:5 eq '_date' ) }
+          {elseif $field.is_datetime_field && $action == 4}
```

Comment:
@eileenmcnaughton for me this is a problem specifically $action == 4 that is what is stopping it from looking right on the confirm page. 

When print out what I think the action maybe it = 1